class LearningError(Exception):
    ...

try:
    raise LearningError("Criando meu pr√≥prio erro")
except LearningError as error:
    error.add_note("Esse erro ocorreu para fins educativos")
    raise 